src/osd/ECUtil.cc

```diff
@@ -9,11 +11,20 @@ using ceph::bufferlist;
 using ceph::ErasureCodeInterfaceRef;
 using ceph::Formatter;
 
+std::pair<uint64_t, uint64_t> ECUtil::stripe_info_t::chunk_aligned_offset_len_to_chunk(
+  std::pair<uint64_t, uint64_t> in) const {
+  return std::make_pair(
+    chunk_aligned_logical_offset_to_chunk_offset(in.first),
+    chunk_aligned_logical_offset_to_chunk_offset(in.second));
```

Comment:
This should be `chunk_aligned_logical_size_to_chunk_size` instead of `chunk_aligned_logical_offset_to_chunk_offset`. The difference is in rounding: the latter already rounds down, the former should round up.

Before switching alignment in the pipeline there was no symptom as `chunk_aligned_offset_len_to_chunk()` was luckily doing the size round up for us.

```cpp
  std::pair<uint64_t, uint64_t> offset_len_to_stripe_bounds(
    std::pair<uint64_t, uint64_t> in) const {
    uint64_t off = logical_to_prev_stripe_offset(in.first);
    uint64_t len = logical_to_next_stripe_offset(
      (in.first - off) + in.second);
    return std::make_pair(off, len);
  }
```

I will push fixup altogether with a unit test.

Reply:
BTW: toggling `osd_ec_partial_reads`  to `false` has tamed the issue above.